docs/ytt-data-values.md

```diff
@@ -37,14 +42,25 @@ Resulting in
 ```yaml
 first: val1
 second: val2
-third: null
+third: val3
 fourth: null
 ```
 
-We typically recommend to use snake case (e.g. `some_key.nested_one`) for naming data values.
-
-Note that if data value contains a `-` (dash) in its name, you will have to use `getattr` function like so `getattr(data.values.key2, "nested-with-dash")` to access its value. This is to avoid parsing ambiguity with substraction operation.
-
+Note:
+- Data Values keys must be strings.
+- use snake-case for keys (e.g. `db_conn.secure`) — keys that contain a `-` (dash)
```

Comment:
I'd still mark this as a recommendation. As in,
We recommend using snake-case...,
to avoid the idea that this is a requirement.

Reply:
Fair enough. Even more relevant with the support for index notation.